### PR TITLE
EW DVR: fix issues related to hosted ports

### PR DESCRIPTION
**Pull request by huntxu**
Hosted ports may not be ready when configuring rules. With this commit a
chance is given for the rules of hosted ports to be re-configured during
the next sync upon such case.

This commit also fix a minor issue when deleting OpenFlow rules of
removed hosted ports.

Fixes: 56cf4bd93 ("openvswitch-agent: implement EW DVR using OpenFlow rules")
Fixes: redmine #10558

Signed-off-by: Hunt Xu <mhuntxu@gmail.com>

### CODE REVIEW

**Comment by zhaochao** [comment]
+1

File: neutron/plugins/openvswitch/agent/openflow_ew_dvr_agent.py [hunk 1]
```diff
@@ -232,8 +232,14 @@ def _del_flows_to_port(self, gateway_macs, port):
...
     def _add_flows_to_hosted_port(self, port, gateway_mac):
+        if port['host'] != self.host:
+            LOG.debug("Port %s is not hosted by this agent.", port['name'])
+            return
         port_ofno = self.int_br.get_port_ofport(port['name'])
-        if port_ofno == INVALID_OFPORT or port['host'] != self.host:
+        if port_ofno == INVALID_OFPORT:
+            LOG.warning("Port %s is not ready.", port['name'])
+            # Setting port['host'] to None for next sync.
+            port['host'] = None
```
Comment (tc1989tc) [inline neutron/plugins/openvswitch/agent/openflow_ew_dvr_agent.py:242]: 其实这个host 设置为Ｎｏｎｅ　是没有用的把？　看后面的update也没有处理ｈｏｓｔ为ｎｏｎｅ的逻辑　只是为了ｄｅｂｕｇ？

Comment (tc1989tc) [reply neutron/plugins/openvswitch/agent/openflow_ew_dvr_agent.py:242]: 哦　好像有用．．．move into host..

Comment (huntxu) [reply neutron/plugins/openvswitch/agent/openflow_ew_dvr_agent.py:242]: 对，就是为了给那里再重新添加一次对hosted port的规则

**Pull request merged**